src/Datasource/EntityTrait.php

```diff
@@ -499,17 +499,33 @@ public function offsetUnset($offset)
     }
 
     /**
-     * Determines whether a method exists in this class
+     * Fetch accessor method name
+     * Accessor methods (available or not) are cached in $_accessors
      *
-     * @param string $method the method to check for existence
-     * @return bool true if method exists
+     * @param string $property the field name to derive getter name from
+     * @param string $type the accessor type ('get' or 'set')
+     * @return string method name or empty string (no method available)
      */
-    protected function _methodExists($method)
+    protected function _accessor($property, $type = 'get')
     {
-        if (empty(static::$_accessors[$this->_className])) {
-            static::$_accessors[$this->_className] = array_flip(get_class_methods($this));
+        if (!isset(static::$_accessors[$this->_className][$type][$property])) {
```

Comment:
one could argue that storing this in APC(u) in production environment would be even better ?